Mobility screening

alexnguyen0512 opened the pull request:
Added docstrings and renamed fields to follow convention. 
Removed materials store and wi_entry as builder input. 
Get wi from electrodes doc.
*Changed get_migration_graph function call to a method call.*

alexnguyen0512 committed:
Create migration_graph.py

New migration_graph builder from electrodes

alexnguyen0512 committed:
Update migration_graph.py

alexnguyen0512 committed:
Updated builder inputs

Removed materials store and wi_entry as builder inputs. get wi from electrode doc.

alexnguyen0512 committed:
Update migration_graph.py

changed field labels to match convention and added docstrings

alexnguyen0512 committed:
Update migration_graph.py

added self to get_migration_graph call

acrutt commented on emmet-builders/emmet/builders/mobility_screening/migration_graph.py:
```diff
@@ -0,0 +1,66 @@
+            self.wi_entries[new_item["working_ion"]] = min(compatibility.process_entries(
+                mpr.get_entries_in_chemsys([new_item["working_ion"]])),
+                                                           key = lambda x : x.energy_per_atom)
+        mg = self.get_migration_graph(compatibility.process_entries(entries),
```
Here I think we want to provided ltol, stol, and angle_tol from the attributes set by the builder. Since get_migration_graph() has default values set, they cannot by changed by the user based on the values used to initialize the builder.

acrutt commented on emmet-builders/emmet/builders/mobility_screening/migration_graph.py:
```diff
@@ -0,0 +1,66 @@
+        new_item["mg"] = mg
+        return new_item
+
+    def get_migration_graph(self, cses: [ComputedEntry], wi_entry: [ComputedEntry], max_distance_min=0.1, max_distance_max=10, ltol=0.4, stol=0.6, angle_tol=15) -> dict:
```
I could be wrong, but I believe the type for cses should be ComputedStructureEntry? If you only provide computed entries without structures, I don't think MigrationGraph.get_structure_from_entries() will work. Also if cse is short for ComputerStructureEntry, you will want to be consistent with your variable names to reduce confusion.

acrutt commented on emmet-builders/emmet/builders/mobility_screening/migration_graph.py:
```diff
@@ -0,0 +1,66 @@
+        entries = [task_doc.structure_entry for task_doc in task_documents]
+        #get entry is working_ion is new. limit MPRester calls
+        if new_item["working_ion"] not in self.wi_entries.keys():
+            self.wi_entries[new_item["working_ion"]] = min(compatibility.process_entries(
```
Is compatibility defined before it is used here? I think this needs to be added to the init:
`from pymatgen.entries.compatibility import MaterialsProject2020Compatibility`
`compatibility = MaterialsProject2020Compatibility()`

acrutt commented on emmet-builders/emmet/builders/mobility_screening/migration_graph.py:
```diff
@@ -0,0 +1,66 @@
+
+        super().__init__(source=electrodes, target=migration_graphs, **kwargs)
+        self.sources.append(tasks)
+
```
Sorry if this is a repeat comment, but I think it might have gotten lost in the shuffle. Can we add a docstring for describing the builder? For example, for the StructureGroup Builder...

> """
        Aggregate materials entries into sgroups that are topotactically similar to each other.
        This is an incremental builder that makes ensures that each materials id belongs to one StructureGroupDoc document
        Args:
            materials (Store): Store of materials documents that contains the structures
            sgroups (Store): Store of grouped material ids
            query (dict): dictionary to limit materials to be analyzed ---
                            only applied to the materials when we need to group structures
                            the phase diagram is still constructed with the entire set
        """
https://github.com/materialsproject/emmet/blob/main/emmet-builders/emmet/builders/materials/electrodes.py#L93

acrutt reviewed (commented):
Best practice is to also follow code format guidelines like PEP 8 which automatic code formatters can do for you. The one I use is called [black](https://github.com/psf/black)

You can pip install it (`pip install black`) and then run it in the command line via `black path_to_file_to_auto_format`

alexnguyen0512 committed:
Update import statements and added docstring

added docstring and necessary import statements. reformatted with black.
also added test file for this builder

alexnguyen0512 commented:
closed to address comments
